src/System.Linq/src/System/Linq/Take.cs

```diff
@@ -15,7 +15,8 @@ public static IEnumerable<TSource> Take<TSource>(this IEnumerable<TSource> sourc
             if (count <= 0) return new EmptyPartition<TSource>();
             IPartition<TSource> partition = source as IPartition<TSource>;
             if (partition != null) return partition.Take(count);
-            return TakeIterator(source, count);
+            IList<TSource> sourceList = source as IList<TSource>;
+            return sourceList != null ? new ListPartition<TSource>(sourceList, 0, count - 1) : TakeIterator(source, count);
```

Comment:
I agree. `Skip()` ended up with a `?:` from the way its changes happened, and this was analogous and so ended up with it too, but I'll change both of them to the more consistent pattern.